File: ApplicationAlgorithm/test/py/unit10.py
def function(a, b):
   m = len(a)
   n = len(b)
   c = [[0 for _ in range(n)] for _ in range(m)]
   print(c)
   for i in range(m):
      for j in range(n):
         if i == 0 or j == 0:
            c[i][j] = 0
         elif a[i] == b[j]:
            c[i][j] = c[i - 1][j - 1] + a[i]
         else:
            c[i][j] = max(c[i - 1][j], c[i][j - 1])
   print(c)
   return c[m - 1][n - 1]

a = []
b = []
m = int(input("Nhap m = "))
for i in range(m):
   a.append(int(input("a[" + str(i + 1) + "] = ")))
n = int(input("Nhap n = "))
for i in range(n):
   b.append(int(input("b[" + str(i + 1) + "] = ")))
print("Day con co tong lon nhat =", function(a, b))

File: ApplicationAlgorithm/test/cpp/unit10.cpp
#include <iostream>
using namespace std;
const int MAX = 1000;
int m, a[MAX], c[MAX][MAX], b[MAX], n;
int main()
{
    cout << "Nhap m = ";
    cin >> m;
    for (int h = 1; h <= m; h++)
    {
        cout << "a[" << h << "] = ";
        cin >> a[h];
    }
    cout << "Nhap n = ";
    cin >> n;
    for (int h = 1; h <= n; h++)
    {
        cout << "b[" << h << "] = ";
        cin >> b[h];
    }
    for (int h = 0; h <= m; h++)
    {
        for (int k = 0; k <= n; k++)
        {
            if (h == 0 || k == 0)
                c[h][k] = 0;
            else if (a[h] == b[k])
                c[h][k] = c[h - 1][k - 1] + a[h];
            else
                c[h][k] = max(c[h - 1][k], c[h][k - 1]);
        }
    }
    cout << "Day con co tong lon nhat = " << c[m][n];
}